File: projects/comparar/commands/tree.rb
#!/usr/bin/env ruby
require "pathname"

class TreeCommand
  BLUE = "\e[94m"
  RESET = "\e[0m"

  def initialize
    @dir_count = 0
    @file_count = 0
  end

  def run(path = ".")
    root = Pathname.new(path)
    puts "#{BLUE}#{root}#{RESET}"
    visualize(root, "")
    puts "\n#{@dir_count} directories, #{@file_count} files"
  end

  private

  def visualize(path, prefix)
    entries = path.children.sort_by { |e| [e.directory? ? 0 : 1, e.basename.to_s.downcase] }
    
    entries.each_with_index do |entry, index|
      is_last = (index == entries.size - 1)
      connector = is_last ? "└── " : "├── "

      name = entry.directory? ? "#{BLUE}#{entry.basename}#{RESET}" : entry.basename      
      puts "#{prefix}#{connector}#{name}"

      if entry.directory?
        @dir_count += 1
        new_prefix = prefix + (is_last ? "    " : "│   ")
        visualize(entry, new_prefix)
      else
        @file_count += 1
      end
    end
  rescue Errno::EACCES
    puts "#{prefix}└── [Permiso denegado]"
  end
end

TreeCommand.new.run(ARGV[0] || ".")

File: projects/comparar/commands/tree.py
#!/usr/bin/env python3
import pathlib
import sys

class TreeCommand:
    BLUE = "\033[94m"
    RESET = "\033[0m"

    def __init__(self):
        self.dir_count = 0
        self.file_count = 0

    def run(self, directory="."):
        root = pathlib.Path(directory)
        print(f"{self.BLUE}{root.name if root.name else directory}{self.RESET}")
        self._visualize(root, "")
        print(f"\n{self.dir_count} directories, {self.file_count} files")

    def _visualize(self, path, prefix):
        try:
            entries = sorted(path.iterdir(), key=lambda p: (not p.is_dir(), p.name.lower()))
        except PermissionError:
            print(f"{prefix}└── [Permiso denegado]")
            return

        for i, entry in enumerate(entries):
            is_last = (i == len(entries) - 1)
            connector = "└── " if is_last else "├── "
            
            name = f"{self.BLUE}{entry.name}{self.RESET}" if entry.is_dir() else entry.name
            print(f"{prefix}{connector}{name}")

            if entry.is_dir():
                self.dir_count += 1
                new_prefix = prefix + ("    " if is_last else "│   ")
                self._visualize(entry, new_prefix)
            else:
                self.file_count += 1

if __name__ == "__main__":
    target = sys.argv[1] if len(sys.argv) > 1 else "."
    TreeCommand().run(target)